internal/network/manifests_generator.go

```diff
@@ -569,12 +573,20 @@ spec:
 `
 
 func (m *ManifestsGenerator) AddNicReapply(ctx context.Context, log logrus.FieldLogger, c *common.Cluster) error {
-	// Add this manifest only is one of the host is installting on an iSCSI boot drive
+	// Add this manifest only if one of the host is installing on an iSCSI / multiapth + iSCSI boot drive
 	_, isUsingISCSIBootDrive := lo.Find(c.Cluster.Hosts, func(h *models.Host) bool {
-		installationDisk, err := hostutil.GetHostInstallationDisk(h)
+		inventory, err := common.UnmarshalInventory(h.Inventory)
 		if err != nil {
 			return false
 		}
+		installationDisk := hostutil.GetDiskByInstallationPath(inventory.Disks, hostutil.GetHostInstallationPath(h))
+		if installationDisk.DriveType == models.DriveTypeMultipath {
+			for _, disk := range inventory.Disks {
+				if disk.DriveType == models.DriveTypeISCSI && strings.Contains(disk.Holders, installationDisk.Name) {
```

Comment:
This condition is used several times, maybe we should fectorize it in a function? Something like `func GetHoldDisks(allDisks []*models.Disk, disk *models.Disk, driveTypeFilter models.DriveType) []*models.Disk`

Reply:
those loops are not costly we act on data in memory, we can pay that price when it improves readability and maintenance.